libsolidity/ast/Types.cpp

```diff
@@ -1961,6 +2008,26 @@ MemberList::MemberMap FunctionType::nativeMembers(ContractDefinition const*) con
 	}
 }
 
+TypePointer FunctionType::encodingType() const
+{
+	// Only external functions can be encoded, internal functions cannot leave code boundaries.
+	if (m_location == Location::External)
+		// This looks like bytes24, but bytes24 is stored differently on the stack.
+		return shared_from_this();
+	else
+		return TypePointer();
+}
+
+TypePointer FunctionType::interfaceType(bool /*_inLibrary*/) const
+{
+	if (m_location != Location::External && m_location != Location::Internal)
+		return TypePointer();
+	if (m_location != Location::External)
+		return TypePointer();
```

Comment:
The two checks above look redundant (the first one catches a subset of what the second one catches).